tests/webTables.spec.ts

```diff
@@ -15,27 +15,20 @@ test.describe('OWNERS page', () => {
 
     test('Validate owners count of the Madison city', async ({ page }) => {
         await page.waitForSelector('tbody')
-        const allCitizensRows = page.locator('tbody tr', { has: page.locator('td').nth(2) })
-        let allMadisonCitizen = 0
-
-        for (const citizen of await allCitizensRows.all()) {
-            const cellValue = await citizen.locator('td').nth(2).textContent()
-            if (cellValue == 'Madison') {
-                allMadisonCitizen += 1
-            }
-        }
-        expect(allMadisonCitizen).toEqual(4)
+        await expect(page.getByRole('row',  {name: "Madison"})).toHaveCount(4)
     })
 
     test('Validate search by Last Name', async ({ page }) => {
+        await page.waitForSelector('table')
```

Comment:
This wait is not needed, the test will work without it.
The step `await buttonFindOwner.click()` will trigger the wait automatically because the owner located in the table, so Playwright will wait until this locator will be available

Reply:
Hmm, not sure.
I got an error 
<img width="908" alt="Знімок екрана 2024-06-30 о 12 57 57" src="https://github.com/AlexxDk/playwright-homework/assets/38554790/2f4b70e9-e93d-4ed1-9727-7a6f3eddc07d">

Reply:
This happened because the table was not loaded within a 10-second timeout. So the timeout worked as expected, the 10 seconds was just not enough to complete the load.
This is a nuance of the application itself. It is hosted in the cloud, and when nobody uses it, it's in hibernation mode. When you open it for the first time, it takes about 30 seconds for the API to spin up and serve the traffic. After that, it works as normal and responsive.
So if you increase the timeout for your test to 40 seconds or just rerun the test right after the failure (application should be up an running by this moment), the test will work fine.

Reply:
Interesting, okay